contracts/EscrowAndStaking/StakingThales.sol

```diff
@@ -599,12 +685,120 @@ contract StakingThales is IStakingThales, Initializable, ProxyOwned, ProxyReentr
+                periodsOfStaking.mod(AMM_EXTRA_REWARD_PERIODS)
+            ].amount.add(amount);
+        }
+
+        if (msg.sender == thalesRangedAMM) {
+            if (lastThalesRangedAMMUpdatePeriod[account] < periodsOfStaking) {
+                thalesRangedAMMVolume[account][periodsOfStaking.mod(AMM_EXTRA_REWARD_PERIODS)].amount = 0;
+                thalesRangedAMMVolume[account][periodsOfStaking.mod(AMM_EXTRA_REWARD_PERIODS)].period = periodsOfStaking;
+                lastThalesRangedAMMUpdatePeriod[account] = periodsOfStaking;
+            }
+            thalesRangedAMMVolume[account][periodsOfStaking.mod(AMM_EXTRA_REWARD_PERIODS)].amount = thalesRangedAMMVolume[
+                account
+            ][periodsOfStaking.mod(AMM_EXTRA_REWARD_PERIODS)].amount.add(amount);
+        }
+
+        if (msg.sender == exoticBonds) {
+            if (lastExoticMarketsUpdatePeriod[account] < periodsOfStaking) {
+                exoticMarketsVolume[account][periodsOfStaking.mod(AMM_EXTRA_REWARD_PERIODS)].amount = 0;
+                exoticMarketsVolume[account][periodsOfStaking.mod(AMM_EXTRA_REWARD_PERIODS)].period = periodsOfStaking;
+                lastExoticMarketsUpdatePeriod[account] = periodsOfStaking;
+            }
+            exoticMarketsVolume[account][periodsOfStaking.mod(AMM_EXTRA_REWARD_PERIODS)].amount = exoticMarketsVolume[
+                account
+            ][periodsOfStaking.mod(AMM_EXTRA_REWARD_PERIODS)].amount.add(amount);
+        }
+
+        if (msg.sender == sportsAMM) {
+            if (lastSportsAMMUpdatePeriod[account] < periodsOfStaking) {
+                sportsAMMVolume[account][periodsOfStaking.mod(AMM_EXTRA_REWARD_PERIODS)].amount = 0;
+                sportsAMMVolume[account][periodsOfStaking.mod(AMM_EXTRA_REWARD_PERIODS)].period = periodsOfStaking;
+                lastSportsAMMUpdatePeriod[account] = periodsOfStaking;
+            }
+            sportsAMMVolume[account][periodsOfStaking.mod(AMM_EXTRA_REWARD_PERIODS)].amount = sportsAMMVolume[account][
+                periodsOfStaking.mod(AMM_EXTRA_REWARD_PERIODS)
+            ].amount.add(amount);
+        }
+
         emit AMMVolumeUpdated(account, amount);
     }
 
+    /// @notice Merge account to transfer all staking amounts to another account
+    /// @param destAccount to merge into
+    function mergeAccount(address destAccount) external notPaused {
+        require(destAccount != address(0) && destAccount != msg.sender, "Invalid address");
+        require(
+            _calculateAvailableRewardsToClaim(msg.sender) == 0 && _calculateAvailableRewardsToClaim(destAccount) == 0,
+            "Cannot merge, claim rewards on both accounts before merging"
+        );
+        require(
+            !unstaking[msg.sender] && !unstaking[destAccount],
+            "Cannot merge, cancel unstaking on both accounts before merging"
+        );
+
```

Comment:
to add the flag for unactive `mergeAccount()`
`require(mergeAccountEnabled, "Merge account is disabled");`

Reply:
probably better to be in first line 740